### PR TITLE
Limit max memory usage in fuzz testing

### LINKED ISSUE #19395: A query may lead to OOM in Fuzz testing in CI environment if server is build with TSan
```
2021.01.21 21:28:31.096590 [ 84 ] {6bc0adce-30a5-47ff-ba9d-33da61a057cb} <Debug> executeQuery: (from [::1]:59472, using production parser) SELECT groupArrayInsertAt(number, number) FROM (SELECT * FROM system.numbers LIMIT 1048575) GROUP BY number ORDER BY number DESC NULLS FIRST
2021.01.21 21:29:30.000676 [ 121 ] {} <Debug> MemoryTracker: Current memory usage (total): 82.85 GiB.
2021.01.21 21:30:30.000684 [ 121 ] {} <Debug> MemoryTracker: Current memory usage (total): 151.89 GiB.
2021.01.21 21:31:30.000581 [ 121 ] {} <Debug> MemoryTracker: Current memory usage (total): 221.06 GiB.
...
==75==FATAL: ThreadSanitizer: internal allocator is out of memory trying to allocate 0x90 bytes
==75==FATAL: ThreadSanitizer: internal allocator is out of memory trying to allocate 0xe8 bytes
==75==ERROR: ThreadSanitizer: allocator is out of memory trying to allocate 0x578 bytes
==75==ERROR: ThreadSanitizer failed to allocate 0x2000 (8192) bytes of InternalMmapVector (error code: 12)
==75==FATAL: ThreadSanitizer: internal allocator is out of memory trying to allocate 0xb8 bytes
```

https://clickhouse-test-reports.s3.yandex.net/19372/0a155ee2f150d1aaed2c223ea7ec7bb816132ce5/fuzzer_tsan/server.log

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Don't allow Query Fuzzer to raise maximum memory usage too much.
This may fix #19395.

### CODE REVIEW

File: docker/test/fuzzer/query-fuzzer-tweaks-users.xml [hunk 1]
```diff
@@ -11,6 +11,10 @@
                     <max>10</max>
                 </max_execution_time>
 
+                <max_memory_usage>
+                    <max>10GB</max>
```
Comment (qoega) [inline docker/test/fuzzer/query-fuzzer-tweaks-users.xml:15]:
```suggestion
                    <max>10737418240</max>
```

Comment (alexey-milovidov) [reply docker/test/fuzzer/query-fuzzer-tweaks-users.xml]: @qoega `10GB` is expected to work, Misha Korotov added this feature one year ago.

**Comment by alexey-milovidov** [comment]
@qoega I won't merge this PR until `10GB` specification will work in settings constrants.

**Comment by qoega** [comment]
> @qoega I won't merge this PR until `10GB` specification will work in settings constrants.

OK.

**Comment by alexey-milovidov** [comment]
It's 10G or 10Gi, not 10GB.
That's reasonable, because it's applicable for any numbers.

**Comment by qoega** [comment]
> It's 10G or 10Gi, not 10GB.
> That's reasonable, because it's applicable for any numbers.

There is a nice place to improve error text for a user.

**Pull request merged**